notify/src/connection.rs

```diff
@@ -17,20 +17,32 @@ pub trait Connection: Clone + Debug + Send + Sync + 'static {
     fn is_closed(&self) -> bool;
 }
 
+#[derive(Clone, Debug)]
+pub enum ChannelType {
+    Closable,
+    Persistent,
+}
+
 #[derive(Clone, Debug)]
 pub struct ChannelConnection<N>
 where
     N: Notification,
 {
     sender: Sender<N>,
+    channel_type: ChannelType,
 }
 
 impl<N> ChannelConnection<N>
 where
     N: Notification,
 {
-    pub fn new(sender: Sender<N>) -> Self {
-        Self { sender }
+    pub fn new(sender: Sender<N>, channel_type: ChannelType) -> Self {
+        Self { sender, channel_type }
+    }
+
+    /// Close the connection, ignoring the channel type
+    pub fn force_close(&self) -> bool {
```

Comment:
Is this method should be public?
If it uses generic with PhantomData instead of Enum it will be possible to implement this method only for Closable ChannelConnection

Reply:
I think `force_close` will actually be used in the `Persistent` case